src/lib/Imp.hs

```diff
@@ -336,26 +333,23 @@ indexSetSize (TC con) = case con of
       ExclusiveLim x -> indexToInt n x
       Unlimited      -> indexSetSize n
     impSub high' low'
-  RecType r -> do
-    sizes <- traverse indexSetSize r
-    impProd $ toList sizes
+  PairType a b -> do
+    a' <- indexSetSize a
+    b' <- indexSetSize b
+    impMul a' b'
   BaseType BoolType -> return $ IIntVal 2
-  SumType (l, r) -> bindM2 impAdd (indexSetSize l) (indexSetSize r)
+  SumType l r -> bindM2 impAdd (indexSetSize l) (indexSetSize r)
   _ -> error $ "Not implemented " ++ pprint con
 indexSetSize ty = error $ "Not implemented " ++ pprint ty
 
 elemCount :: ScalarTableType -> ImpM IExpr
 elemCount t = case t of
-  TabType pi | isDependentType pi ->
-    error $ "Tables with sizes of dimensions dependent on previous dimensions are not supported: " ++ pprint t
```

Comment:
Nooooo, this is super important! We cannot handle tables with dimensions dependent on other dimensions, and this is our primary safeguard

Reply:
Sorry, my bad. This was from the comment-stuff-out-until-it-compiles phase of my merge.